src/mds/MDSMap.cc

```diff
@@ -643,10 +643,8 @@ void MDSMap::encode(bufferlist& bl, uint64_t features) const
   encode(data_pools, bl);
   encode(cas_pool, bl);
 
-  if (epoch == 0) {
-    ENCODE_FINISH(bl);
+  if (is_client)
```

Comment:
not stop here. I mean skip fields from 'compat' to 'min_compat_client'

Reply:
@ukernel I'm confused with this comment as there are multiple compat fields. Do you mean skip the new ev part from 'compat' to 'min_compat_client' ? The below one?

```
    // ev part                                                                                                                                                                                                                                                                  
    ENCODE_START(1, 1, bl);                                                                                                                                                                                                                                                     
    encode(compat, bl);                                                                                                                                                                                                                                                         
    encode(metadata_pool, bl);                                                                                                                                                                                                                                                  
    encode(tableserver, bl);                                                                                                                                                                                                                                                    
    encode(ever_allowed_features, bl);                                                                                                                                                                                                                                          
    encode(explicitly_allowed_features, bl);                                                                                                                                                                                                                                    
    encode(balancer, bl);                                                                                                                                                                                                                                                       
    encode(standby_count_wanted, bl);                                                                                                                                                                                                                                           
    encode(old_max_mds, bl);                                                                                                                                                                                                                                                    
    encode(min_compat_client, bl); 
    ENCODE_FINISH(bl);                                                                                                                                                                                                                                                          
```

Reply:
yes

Reply:
Okay, so in this particular case of client, there would be 0 encodings in the old ev part, without the corresponding field in the new ev part. Is that correct?

```
  encode((int64_t)0, bl);                                                                                                                                                                                                                                                       
  encode(created, bl);                                                                                                                                                                                                                                                          
  encode(modified, bl);                                                                                                                                                                                                                                                         
  encode((int32_t)0, bl);
```

Reply:
no. you should update MDSMap::decode instead here